docs/docs/clients/01-typescript.md

```diff
@@ -1,265 +1,365 @@
+
+  optimizeDeps: {
+    esbuildOptions: {
+      define: {
+        global: "globalThis",
+      },
+      plugins: [
+        NodeGlobalsPolyfillPlugin({
+          buffer: true,
+        }),
+      ],
+    },
+  },
+});
 ```
 
-You are then ready to use the generated client code inside this project directly or by publishing the client and
-installing it as any other npm package.
+You are then ready to use the generated client code inside this project directly
+or by publishing the client and installing it as any other `npm` package.
 
-## Usage
+After the chain starts, you will see Frank's address is
+`cosmos13xkhcx2dquhqdml0k37sr7yndquwteuvt2cml7`. We'll be using Frank's account
+for querying data and broadcasting transactions in the next section.
 
-The code generated in `ts-client` comes with a `package.json` file ready to publish which you can modify to suit your
-needs.
+## Querying
 
-The client is based on a modular architecture where you can configure a client class to support the modules you need
-and instantiate it.
+The code generated in `ts-client` comes with a `package.json` file ready to
+publish which you can modify to suit your needs. To use`ts-client` install the
+required dependencies:
 
-By default, the generated client exports a client class that includes all the Cosmos SDK, custom and 3rd party modules
-in use in your project.
+```
+cd ts-client
+npm install
+```
 
-To instantiate the client you need to provide environment information (endpoints and chain prefix) and an optional
-wallet (implementing the CosmJS OfflineSigner interface).
+The client is based on a modular architecture where you can configure a client
+class to support the modules you need and instantiate it.
 
-For example, to connect to a local chain instance running under the Ignite CLI defaults, using a CosmJS wallet:
+By default, the generated client exports a client class that includes all the
+Cosmos SDK, custom and 3rd party modules in use in your project.
 
-```typescript
-import { Client } from '<path-to-ts-client>';
-import { DirectSecp256k1HdWallet } from "@cosmjs/proto-signing";
+To instantiate the client you need to provide environment information (endpoints
+and chain prefix) and an optional wallet (implementing the CosmJS
+`OfflineSigner` interface).
 
-const mnemonic = "surround miss nominee dream gap cross assault thank captain prosper drop duty group candy wealth weather scale put";
-const wallet = await DirectSecp256k1HdWallet.fromMnemonic(mnemonic);
+For example, to connect to a local chain instance running under the Ignite CLI
+defaults, using a CosmJS wallet:
 
-const client = new Client({ 
-		apiURL: "http://localhost:1317",
-		rpcURL: "http://localhost:26657",
-		prefix: "cosmos"
-	},
-	wallet
+```typescript title="my-frontend-app/src/main.ts"
+import { Client } from "../../ts-client";
+
+const client = new Client(
+  {
+    apiURL: "http://localhost:1317",
+    rpcURL: "http://localhost:26657",
+    prefix: "cosmos",
+  },
+  wallet
```

Comment:
wallet here is undefined, you should reintroduce the lines for CosmJS wallet setup (albeit using the mnemonic from `config.yml` above)

Reply:
or remove wallet altogether if only setting up a querying client

Reply:
https://github.com/ignite/cli/pull/3192/commits/b07cbbea24029b55e2160d0adcd79c270c10a0df